tools/src/tensil/tools/compiler/OnnxFrontend.scala

```diff
@@ -193,15 +193,16 @@ class OnnxFrontend(
       .map(outputNodeNames(_))
       .flatten
       .distinct
-      .foldLeft(prev)((prev, nodeName) => {
-        recursiveTraverse(
-          nodeName +: prev,
-          nodeProtos(nodeName).input.toSet
-            .diff(tensorProtos.keySet ++ inputValueInfoProtos.keySet)
-            .toSeq
-        )
-      })
-      .distinct
+      .foldLeft(prev)((prev, nodeName) =>
+        if (!prev.contains(nodeName))
+          recursiveTraverse(
+            prev,
+            nodeProtos(nodeName).input.toSet
+              .diff(tensorProtos.keySet ++ inputValueInfoProtos.keySet)
+              .toSeq
+          ) :+ nodeName
```

Comment:
This appears to reverse the order of results, is that intentional? if you did `nodeName +: recursiveTraverse(...)` instead it would preserve the order

Reply:
Since we start the traversal from the output nodes the reversed order will give us inputs-to-outputs sequence, which is what we want.